kornia/augmentation/functional.py

```diff
@@ -204,31 +231,30 @@ def apply_hflip(input: torch.Tensor, params: Dict[str, torch.Tensor], return_tra
     input = _transform_input(input)
     _validate_input_dtype(input, accepted_dtypes=[torch.float16, torch.float32, torch.float64])
 
-    if not isinstance(return_transform, bool):
-        raise TypeError(f"The return_transform flag must be a bool. Got {type(return_transform)}")
-
     flipped: torch.Tensor = input.clone()
 
     to_flip = params['batch_prob'].to(input.device)
     flipped[to_flip] = hflip(input[to_flip])
 
-    if return_transform:
-
-        trans_mat: torch.Tensor = torch.eye(3, device=input.device, dtype=input.dtype).repeat(input.shape[0], 1, 1)
+    return flipped
 
-        w: int = input.shape[-1]
-        flip_mat: torch.Tensor = torch.tensor([[-1, 0, w],
-                                               [0, 1, 0],
-                                               [0, 0, 1]])
 
-        trans_mat[to_flip] = flip_mat.type_as(input)
+def compute_hflip_transformation(input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
+    input = _transform_input(input)
+    _validate_input_dtype(input, accepted_dtypes=[torch.float16, torch.float32, torch.float64])
```

Comment:
we should also abstract this to kornia.testing to provide an API for checking classes types, shapes, dtype and devices):
```python
assert_is_tensor(my_tensor)  # check that is torch.Tensor
assert_eq_shape(my_tensor, (B, CH, D, H, W))  # check that shape is as provided
assert_eq_dtype(my_tensor, (torch.float32, torch.float64)) # check all possible tested dtypes 
assert_eq_device(my_tensor, my_device)  # check tensor is in a specific device
```

@lferraz had some ideas for this too